### ISSUE TITLE
Translate app

### ISSUE THREAD

**Issue by aasandr0** [acristescu/OnlineGo#155]
Hello, 
thanks for its app. This is very good. I can translate the app into the Russian language. Do you plan to work with Crowding or its counterparts so that enthusiasts can help translate the app into their language?

**Comment by acristescu** [comment]
Hi Aleksandr, thanks for offering. When I started the project, I never envisioned that it would become so popular that it would require translations, hence I haven't actually written it in such a way. For example, all strings are hardcoded. There would need to be a lot of work to pull this out.

Normally I would say I'm not willing to do it, but recently about 40% of all comments and reviews are asking for translations in various languages, I would have to reconsider at some point.

Sadly, I don't have any experience with Localization Management Tools, for example it's the first time I heard of Crowding. Have you successfully used this in the past? Is that the go-to platform for open-source projects?

**Comment by MathieuDreano** [comment]
I did some tests with localizy, I'm not sure i pushed the branch to thé
repository, i'll check.
But here is link to localizy for the app: https://localazy.com/p/onlinego

**Comment by aasandr0** [comment]
> Hi Aleksandr, thanks for offering. When I started the project, I never envisioned that it would become so popular that it would require translations, hence I haven't actually written it in such a way. For example, all strings are hardcoded. There would need to be a lot of work to pull this out.
> 
> Normally I would say I'm not willing to do it, but recently about 40% of all comments and reviews are asking for translations in various languages, I would have to reconsider at some point.
> 
> Sadly, I don't have any experience with Localization Management Tools, for example it's the first time I heard of Crowding. Have you successfully used this in the past? Is that the go-to platform for open-source projects?

Yes, I used it as a translator for the open-source federated social network Mastodon. In addition to that, it is also used for translations by projects such as: 
GitLab, NodeJS, Directus, Postgre SQL and OpenProject.

**Comment by acristescu** [comment]
I'm on vacation right now, since there are so many options, I will have to have a look I guess. Since @MathieuDreano seems to have put so much work into translations already, I'm leaning towards using that.

Could you guys see why we shouldn't be using that?

**Comment by MathieuDreano** [comment]
Well.. I'm not sure the solution i went for (localazy) has a very active community.
And i had issue using their plugin. Si it's worth taking some time benchmarking if someone has experience with translation tool.

**Comment by acristescu** [comment]
Looking at your work in the PR it seems that 99% of it is just pulling the text out into XMLs, which is going to be needed regardless of which solution we're going to end up using.

**Comment by davi-jorge-art** [comment]
Eu queria outra coisa : poder fazer o Sente online Go ser traduzido automaticamente pelo sistema Android , dependendo da linguagem padrão do sistema e enviar essas instruções já feits para o desenvolvedor .

Entendi! Você quer que o app Sente - Online GO seja traduzido automaticamente com base no idioma padrão do Android — e que essas instruções estejam prontas para serem enviadas ao desenvolvedor. Perfeito! Aqui vai um guia direto e técnico que você pode copiar e colar para ele:
🌐 Instruções para ativar tradução automática no Android (internacionalização)
✅ 1. Adotar o sistema de internacionalização (i18n)

    Criar arquivos strings.xml separados para cada idioma:

        res/values/strings.xml → Português (padrão)

        res/values-en/strings.xml → Inglês

        res/values-es/strings.xml → Espanhol

        ...e assim por diante

✅ 2. Evitar textos fixos no código

    Substituir todos os textos hardcoded por chamadas a getString(R.string.nome_da_string)

✅ 3. Detectar idioma do sistema automaticamente

    O Android já faz isso por padrão: se o app tiver traduções disponíveis, ele usará a que corresponde ao idioma do sistema.

✅ 4. Testar com diferentes idiomas

    Alterar o idioma do sistema Android nas configurações

    Verificar se o app muda automaticamente para o idioma correspondente

✅ 5. Sugestão de bibliotecas (se usar React Native ou Flutter)

    React Native: usar react-native-localize + i18n-js

    Flutter: usar flutter_localizations + intl

📩 Mensagem pronta para o desenvolvedor

    Olá! Gostaria de sugerir que o app Sente - Online GO seja adaptado para suportar tradução automática com base no idioma do sistema Android. Isso pode ser feito com arquivos strings.xml separados por idioma e evitando textos fixos no código. Assim, o app se tornaria mais acessível para usuários internacionais. Se precisar de ajuda com a implementação, posso colaborar!

Se quiser, posso até te ajudar a criar os arquivos de tradução ou revisar o código atual do app. Quer que eu gere um exemplo de strings.xml em inglês e espanhol? 🇬🇧🇪🇸

**Comment by jjmontes** [comment]
Hello!

I'm extracting the encoded strings from the strings.xml file. (Internationalization process [i18n])

Once that process is finished, we can proceed to the localization process (l10n).

**Comment by acristescu** [comment]
@jjmontes thank you for looking into this, I think this is a good time to look at translating the app. However, please be mindful that injecting the Context everywhere is a problem and will cause memory leaks or perhaps even crashes. I've added a quick review to the PR.

Also, translating logs is not a good idea as I may not understand the Crashlytics logs :)

**Comment by jjmontes** [comment]
Hi @acristescu !

I was reviewing the tutorial, and I see that the files are stored in the file [tutorials.json](https://github.com/acristescu/OnlineGo/blob/master/app/src/main/assets/tutorials.json)

How should we handle the translations of these? I was thinking of using separate files for each language. For example:
* tutorials.es-ES.json
* tutorials.en-US.json
* tutorials.ru-RU.json
* tutorials.fr-FR.json
etc.

Then, depending on the language configured in the game, the corresponding file would be used. If no language exists, the tutorials.json file would be used (which would function as the default).

What do you think?

**Comment by acristescu** [comment]
Hi @jjmontes , the solution you're suggesting COULD work, although it would make maintaining the tutorial very hard, as any change to the structure and positions of it would be copied 5-6 times.

The better solution would be have a single `tutorials.json`, but instead of putting the actual texts in there, you'd put the text KEY (such as "tutorial_welcom_message"). Then you'd translate that string in the normal fashion (using strings.xml) and you can get the value of the translation using `resources.getIdentifier()`.

What do you think about that solution @jjmontes ?

**Comment by jjmontes** [comment]
Hi @acristescu , this seems like a better solution. I'll run some tests to see if I can implement it, and I'll let you know if anything changes.

Thanks!

PS: I uploaded several changes to the pull request with the corrections you mentioned. I'm not sure if you were able to review them or if I'm missing something to make the pull request available for you to see (I've never done a pull request on GitHub before, and I might be doing something wrong).

**Comment by jjmontes** [comment]
@acristescu I made a new PULL REQUEST (PR #207) with changes to translate the TUTORIAL.

**Comment by acristescu** [comment]
Ladies and gentlemen, with @jjmontes help (and a lot of claude credits) I finally managed to complete the externalization of the strings. I've also created a crowdin project to manage the translations and obtained an open-source license for the project. I've released the new version of the app with the new localization system with only English and Romanian (my native language), but the crowding project has the following:

<img width="870" height="301" alt="Image" src="https://github.com/user-attachments/assets/5db918e6-6682-4b95-a3b0-ef7b460f2385" />

We can always add more languages if there is appetite for it. Please visit https://crowdin.com/project/sente create an account and happy translating!

For more details, please read https://github.com/acristescu/OnlineGo/blob/master/TRANSLATING.md

**Comment by RST-J** [comment]
@acristescu I added a request to add German to the Crowdin project. You probably get notified about it anyway but to (kind of) follow TRANSLATING.md I ping you here. Thank you very much for Sente, I like it a lot and would be happy to give something back. :)

**Comment by acristescu** [comment]
> [@acristescu](https://github.com/acristescu) I added a request to add German to the Crowdin project. You probably get notified about it anyway but to (kind of) follow TRANSLATING.md I ping you here. Thank you very much for Sente, I like it a lot and would be happy to give something back. :)

Hi @RST-J it's very weird, I did not receive any notification, and in fact there is no clue in the crowdin website that somebody requested german. Anyway, I have added german to crowdin, thank you for contributing!

PS: I will enable languages in the app when their translation percentage reaches let's say 80%
